compiler/rustc_data_structures/src/sync.rs

```diff
@@ -367,31 +367,37 @@ cfg_if! {
             }
         }
 
-        pub use rayon::scope;
+        pub fn scope<'scope, OP, R>(op: OP) -> R
+        where
+            OP: FnOnce(&rayon::Scope<'scope>) -> R + DynSend,
+            R: DynSend,
+        {
+            let op = FromDyn::from(op);
+            rayon::scope(|s| FromDyn::from(op.into_inner()(s))).into_inner()
+        }
 
         /// Runs a list of blocks in parallel. The first block is executed immediately on
         /// the current thread. Use that for the longest running block.
         #[macro_export]
         macro_rules! parallel {
-            (impl $fblock:block [$($c:expr,)*] [$block:expr $(, $rest:expr)*]) => {
-                parallel!(impl $fblock [$block, $($c,)*] [$($rest),*])
+            ($fblock:block [$($c:expr,)*] [$block:expr $(, $rest:expr)*]) => {
+                parallel!($fblock [$block, $($c,)*] [$($rest),*])
             };
-            (impl $fblock:block [$($blocks:expr,)*] []) => {
+            ($fblock:block [$($blocks:expr,)*] []) => {
                 {
-                    let fblock = rustc_data_structures::sync::FromDyn::from(|| $fblock);
-                    let blocks = parallel!(tuple [$($blocks,)*]);
-                    ::rustc_data_structures::sync::scope(|s| {
-                        parallel!(spawn blocks, s, [$($blocks,)*]);
-                        fblock.into_inner()();
-                    })
+                    scope(|s| {
```

Comment:
Use `::rustc_data_structures::sync::scope` to avoid the import in other crates.

Reply:
That makes sense! Done :)